tomviz/PipelineModel.cxx

```diff
@@ -685,8 +686,9 @@ void PipelineModel::dataSourceAdded(DataSource* dataSource)
   beginInsertRows(QModelIndex(), 0, 0);
   m_treeItems.append(treeItem);
   endInsertRows();
-  connect(dataSource, SIGNAL(operatorAdded(Operator*)),
-          SLOT(operatorAdded(Operator*)));
+  dataSource->pipeline()->disconnect(this);
```

Comment:
I think you can avoid this ```disconnect(...)``` by using ```Qt::UniqueConnection``` when connecting the signal, it will only make one connection.